src/memory.rs

```diff
@@ -0,0 +1,63 @@
+use std::cell::Cell;
+use rand::rngs::StdRng;
+
+use rustc_mir::interpret::{Pointer, Allocation, AllocationExtra, InterpResult};
+use rustc_target::abi::Size;
+
+use crate::{stacked_borrows, intptrcast};
+use crate::stacked_borrows::{Tag, AccessKind};
+
+#[derive(Default, Clone, Debug)]
+pub struct MemoryState {
+    pub stacked: stacked_borrows::MemoryState,
+    pub intptrcast: intptrcast::MemoryState,
+    /// The random number generator to use if Miri
+    /// is running in non-deterministic mode
+    pub(crate) rng: Option<StdRng>
+}
+
+#[derive(Debug, Clone,)]
+pub struct AllocExtra {
+    pub stacks: stacked_borrows::Stacks,
+    pub base_addr: Cell<Option<u64>>,
```

Comment:
Please make this `stacked_borrows::AllocExtra` and `intptrcast::AllocExtra`, for consistency with `MemoryState`.
Stacked Borrows can have a `pub type AllocExtra = Stacks;` for this.